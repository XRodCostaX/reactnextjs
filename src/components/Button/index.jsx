import P from 'prop-types';
import { Component } from 'react';
import { Container } from './styled';

export default class Button extends Component {
  render() {
    const { text, onClick, disabled } = this.props;

    return (
      <Container>
        <button disabled={disabled} className="button" onClick={onClick}>
          {text}
        </button>
      </Container>
    );
  }
}
Button.defaultProps = {
  disable: false,
};
Button.propTypes = {
  text: P.string.isRequired,
  onClick: P.func,
  disable: P.bool,
};
