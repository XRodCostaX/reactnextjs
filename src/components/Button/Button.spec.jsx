/* eslint-disable react/jsx-boolean-value */
import { fireEvent, render, screen } from '@testing-library/react';
import { vi } from 'vitest';
import Button from '.';

describe('<Button />', () => {
  it('should render the button with the text "load more"', () => {
    const fn = vi.fn();
    render(<Button text="Load more" disabled={false} onClick={fn} />);
    expect.assertions(1);

    const button = screen.getByRole('button', { name: /load more/i });
    expect(button).toBeInTheDocument();
  });

  it('should call a function on button click', () => {
    const fn = vi.fn();
    render(<Button text="Load more" disabled={false} onClick={fn} />);

    const button = screen.getByRole('button', { name: /load more/i });

    fireEvent.click(button);

    expect(fn).toHaveBeenCalledTimes(1);
  });

  it('should be disable when disable is true', () => {
    const fn = vi.fn();
    render(<Button text="Load more" disabled={true} onClick={fn} />);

    const button = screen.getByRole('button', { name: /load more/i });
    expect(button).toBeDisabled();
  });

  it('should be enabled when disable is false', () => {
    const fn = vi.fn();
    render(<Button text="Load more" disabled={false} onClick={fn} />);

    const button = screen.getByRole('button', { name: /load more/i });

    expect(button).toBeEnabled();
  });
  it('should match snapshot', () => {
    const { container } = render(<Button text="Load more" disable />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
