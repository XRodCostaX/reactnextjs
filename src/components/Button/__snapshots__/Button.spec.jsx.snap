// Vitest Snapshot v1

exports[`<Button /> > should match snapshot 1`] = `
<div
  class="sc-bcXHqe iSyUUh"
>
  <button
    class="button"
  >
    Load more
  </button>
</div>
`;
